api/seccompprofile/v1beta1/seccompprofile_types.go

```diff
@@ -183,3 +212,30 @@ type SeccompProfileList struct {
 func init() { //nolint:gochecknoinits // required to init scheme
 	SchemeBuilder.Register(&SeccompProfile{}, &SeccompProfileList{})
 }
+
+func UnionSyscalls(baseSyscalls, appliedSyscalls []*Syscall) []*Syscall {
+	allSyscalls := make(map[seccomp.Action]map[string]bool)
```

Comment:
uhm... you already have a notion of how long the syscall lists will be from the two objects being passed. Why not directly allocate this map with the length of the longest list of the two? That way you reduce the number of allocations made at runtime.

Reply:
I can fix these comments (they are valid) just note that I just lifted the existing function from the seccomp controller :-) and didn't write it from scratch.

Reply:
@jhrozek ah! I hadn't noticed.

Reply:
Done in a separate commit.